src/lem/README.md

```diff
@@ -0,0 +1,94 @@
+## Lurk Evaluation Model (LEM)
+
+The main purpose of LEM is to represent Lurk's step function *as data*, which can
+be interpreted or used to synthesize the circuit.
+
+### Pointer cycle
+
+The variables of a LEM are represented by *meta pointers*, which become bound to
+concrete pointers during interpretation. So a `MetaPtr` is a named reference to
+a `Ptr`.
+
+To synthesize the circuit, the `Ptr`s that come from interpretation are hashed
+and become `ZPtr`s, whose data can be allocated in the circuit to generate
+*allocated pointers* (`AllocatedPtr`).
+
+```mermaid
+graph TD;
+    M[MetaPtr] --> P[Ptr];
+    P --> Z[ZPtr];
+    Z --> A[AllocatedPtr];
+```
+
+### Check
+
+We statically check the following properties:
+
+* Static single assignments
```

Comment:
This implies that the LEM author must write SSA directly. Is that right?

Reply:
Yes